core/unix/os_exports.h

```diff
@@ -112,6 +115,17 @@
 # error NYI
 #endif
 
+#if defined(MACOS) && defined(X64)
+/* FIXME: current pthread_t struct has the first TLS entry at offset 28. We should
+ * provide a dynamic method to determine the first entry for forward compatability.
+ * Starting w/ libpthread-218.1.3 they now leave slots 6 and 11 unused to allow
+ * limited interoperability w/ code targeting the Windows x64 ABI. We steal slot 6
+ * for our own use.
```

Comment:
I could see two approaches: one, we go the Windows-ish route initially (so #3) and steal several TLS slots from the app and use those, intending to tweak later once we have early injection.  This lets us make faster progress toward being able to run 64-bit apps under simple tools.  For larger (e.g., Dr. Memory) or even simple library-using tools (e.g., written in C++), we will run into isolation issues and eventually want the private loader, but we can first finish the other issues and have an end-to-end 64-bit system working reasonably well.

Or, we take a longer view and get the private loader working now.  This would help 32-bit as well as the code will presumably be very similar, and by starting on a 32-bit private loader it can be easily tested end-to-end on apps and tools.  We have known, concrete cases where Dr. Memory failed due to lack of private loader isolation, which can be used as initial test cases.

I would say it's up to you.  (I vote for the private loader but could see it either way.)